scripts/validate_data.py

```diff
@@ -1,74 +1,79 @@
-""" Python script to validate data
-
-Run as:
-
-    python3 scripts/validate_data.py data
-"""
-
 from pathlib import Path
 import sys
 import hashlib
 
-
 def file_hash(filename):
-    """ Get byte contents of file `filename`, return SHA1 hash
-
+    """ Get SHA1 hash of the contents of a file
     Parameters
     ----------
     filename : str
         Name of file to read
-
     Returns
     -------
     hash : str
         SHA1 hexadecimal hash string for contents of `filename`.
     """
-    # Open the file, read contents as bytes.
-    # Calculate, return SHA1 has on the bytes from the file.
-    # This is a placeholder, replace it to write your solution.
-    raise NotImplementedError(
-        'This is just a template -- you are expected to code this.')
-
+    sha1 = hashlib.sha1()
+    with open(filename, 'rb') as fobj:
+        while True:
+            data = fobj.read(65536)  # Read in 64k chunks
```

Comment:
It's relatively rare that you'll run out of memory for this check - the image would have to be huge.   And I think you'd find in that case that your code would be unacceptably slow.

Reply:
You are correct. Thanks for correcting me on this I will update the code

Reply:
DId you intend to change this code?